### PR TITLE
fix: fix failed hermit binary source tests due to findUp module acts differently in windows

### PR DESCRIPTION
**Pull request by lyonlai**
## Changes
This PR fixes failed hermit binary source tests. It uses the same way as how `findUpLocal` tests sets up the `find-up` mocks in [`lib/util/fs/index.spec.ts`](https://github.com/renovatebot/renovate/blob/main/lib/util/fs/index.spec.ts#L367)

The failed tests result is in the following.
https://github.com/renovatebot/renovate/runs/7685608162?check_suite_focus=true#step:7:306

## Context

This happens after the merge of https://github.com/renovatebot/renovate/pull/16259/

## Documentation (please check one with an [x])

- [ ] I have updated the documentation, or
- [X] No documentation update is required

## How I've tested my work (please tick one)

I have verified these changes via:

- [ ] Code inspection only, or
- [X] Newly added/modified unit tests, or
- [ ] No unit tests but ran on a real repository, or
- [ ] Both unit tests + ran on a real repository

### CODE REVIEW

**Comment by lyonlai** [comment]
@viceice please take a look. and also please label this PR with `ci:fulltest` to match the test in `main` branch

**Review by rarkins — DISMISSED** [review]

File: lib/util/exec/hermit.spec.ts [hunk 1]
```diff
@@ -16,42 +21,37 @@ describe('util/exec/hermit', () => {
...
 
-    afterEach(async () => {
-      await localDirResult?.cleanup();
-    });
-
     it('should find the closest hermit cwd to the given path', async () => {
-      await writeLocalFile('nested/bin/hermit', 'foo');
-      await writeLocalFile('bin/hermit', 'bar');
-
+      findUp.mockResolvedValue(upath.join(localDir, 'nested/bin/hermit'));
```
Comment (viceice) [inline lib/util/exec/hermit.spec.ts:29]:
```suggestion
      findUp.mockResolvedValueOnce(upath.join(localDir, 'nested/bin/hermit'));
```
use once to auto reset the return value of the mock

File: lib/util/exec/hermit.spec.ts [hunk 2]
```diff
@@ -16,42 +21,37 @@ describe('util/exec/hermit', () => {
...
       const nestedCwd = 'nested/other/directory';
 
       expect(await findHermitCwd(nestedCwd)).toBe(`${localDir}/nested/bin`);
       expect(await findHermitCwd('nested')).toBe(`${localDir}/nested/bin`);
+
+      findUp.mockResolvedValue(upath.join(localDir, 'bin/hermit'));
```
Comment (viceice) [inline lib/util/exec/hermit.spec.ts:35]:
```suggestion
      findUp.mockResolvedValueOnce(upath.join(localDir, 'bin/hermit'));
```

File: lib/util/exec/hermit.spec.ts [hunk 3]
```diff
@@ -16,42 +21,37 @@ describe('util/exec/hermit', () => {
...
       expect(await findHermitCwd('')).toBe(`${localDir}/bin`);
       expect(await findHermitCwd('other/directory')).toBe(`${localDir}/bin`);
     });
 
     it('should throw error when hermit cwd is not found', async () => {
       const err = new Error('hermit not found for other/directory');
+      findUp.mockResolvedValue('');
```
Comment (viceice) [inline lib/util/exec/hermit.spec.ts:42]: i think you can remove this

File: lib/util/exec/hermit.spec.ts [hunk 4]
```diff
@@ -16,42 +21,37 @@ describe('util/exec/hermit', () => {
...
   });
 
   describe('getHermitEnvs', () => {
+    beforeEach(() => {
+      GlobalConfig.set({ localDir });
+    });
+
     it('should return hermit environment variables when hermit env returns successfully', async () => {
-      await writeLocalFile('bin/hermit', 'bar');
+      findUp.mockResolvedValue(upath.join(localDir, 'bin/hermit'));
```
Comment (viceice) [inline lib/util/exec/hermit.spec.ts:54]:
```suggestion
      findUp.mockResolvedValueOnce(upath.join(localDir, 'bin/hermit'));
```

**Review by viceice — CHANGES_REQUESTED** [review]

**Review by viceice — APPROVED** [review]

**Pull request merged**